.gitlab-ci.yml

```diff
@@ -63,142 +73,93 @@ checks:
-    - bin/build
-    - BINARY=build/myst/myst bin/package_debian $BUILD_VERSION amd64
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageLinuxDebianAmd64
 
 package:linux-debian-arm:
   stage: build
-  tags:
-    - go
-  script:
-    - go get -u github.com/debber/debber-v0.3/cmd/debber
-    - bin/build_xgo linux/arm
-    - BINARY=build/myst/myst_linux_arm bin/package_debian $BUILD_VERSION armhf
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageLinuxDebianArm
 
 package:linux-raspberry-image:
   stage: build
-  tags:
-    - go
-  script:
-    - go get -u github.com/debber/debber-v0.3/cmd/debber
-    - bin/build_xgo linux/arm
-    - BINARY=build/myst/myst_linux_arm bin/package_debian $BUILD_VERSION armhf
-    - bin/package_raspberry
-  only:
-    - tags
+  tags: [go]
+  script: go run mage.go -v PackageLinuxRaspberryImage
 
 package:osx-amd64:
   stage: build
-  tags:
-    - go
-  script:
-    - BINARY=build/myst/myst_darwin_amd64 bin/package_standalone darwin amd64
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageOsxAmd64
 
 package:windows-amd64:
   stage: build
-  tags:
-    - go
-  script:
-    - BINARY=build/myst/myst_windows_amd64.exe bin/package_standalone windows amd64
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageWindowsAmd64
 
 package:ios:
   stage: build
-  tags:
-    - go
-  script:
-    - bin/package_ios amd64
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageIOS
 
 package:android:
   stage: build
-  tags:
-    - go
-  script:
-    - bin/package_android amd64
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageAndroid
 
 package:docker-alpine:
   stage: build
-  tags:
-    - go
-  script:
-    - bin/package_docker
-    - docker save myst:alpine | gzip > myst_alpine.tgz
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageDockerAlpine
 
 package:docker-ubuntu:
   stage: build
-  tags:
-    - go
-  script:
-    - bin/package_docker_ubuntu $BUILD_VERSION
-    - docker save myst:ubuntu | gzip > myst_ubuntu.tgz
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageDockerUbuntu
 
 package:swagger-redoc-docker:
   stage: build
-  tags:
-    - go
-  script:
-    - go get -u github.com/go-swagger/go-swagger/cmd/swagger/
-    - bin/swagger_generate
-    - bin/package_docker_docs
-    - docker save tequilapi:${BUILD_VERSION} | gzip > tequilapi_redoc.tgz
-  only:
-    - master
+  tags: [go]
+  script: go run mage.go -v PackageDockerSwaggerRedoc
 
 release:snapshot:
   stage: release
-  tags:
-    - go
-  script:
-    - echo $BUILD_VERSION
-#  only:
-#    refs:
-#      - master
+  tags: [go]
+  script: go run mage.go -v ReleaseSnapshot
+
+release:goreport:
+  stage: release
+  tags: [go]
+  script: bin/release_goreport
+  only:
+    - master
+
+remove-bucket:
+  stage: post-release
+  tags: [go]
+  script: go run mage.go -v RemoveBucket
```

Comment:
What if the process fails on the early stage? Will we remove the bucked in any case? 
Just to make sure that we are not leaving not cleaned bucket. We had an `after_fail` job on Travis to clean it up.

Reply:
Not sure about this one. On one hand, if it fails in `package` stage, it will leave a bucket behind. On the other, a failed pipeline can be retried. If we remove bucket `on_failure`, we lose this functionality, because then it will try to package and upload to an already deleted bucket.

Reply:
I'm worried about disk space on S3. We will definitely have a lot of failed jobs, because of it's CI.
After some time we will have a bunch of uncleaned buckets on S3.

I'm not sure about retries, in most cases, you are doing another commit to fix a problem and trigger another build. A retried job with no changes will fail again.